hal/src/rtc/rtic/mod.rs

```diff
@@ -0,0 +1,335 @@
+//!     # let mut mclk = unsafe { core::mem::transmute(()) };
+//!     # let mut osc32kctrl = unsafe { core::mem::transmute(()) };
+//!     // Here the RTC clock source should be configured using the clocks API
+//!
+//!     // Start the monotonic
+//!     Mono::start(rtc);
+//! }
+//!
+//! async fn usage() {
+//!     loop {
+//!          // Use the monotonic
+//!          let timestamp = Mono::now();
+//!
+//!          Mono::delay_until(timestamp + 2u32.secs()).await;
+//!          Mono::delay(100u32.millis()).await;
+//!     }
+//! }
+//! ```
+//!
+//! # Other notes
+//!
+//! The number returned by
+//! [`Monotonic::now().ticks()`](rtic_monotonic::Monotonic::now) will always
+//! increase (barring monotonic rollover). However, due to the register
+//! [synchronization delay](https://onlinedocs.microchip.com/oxy/GUID-F5813793-E016-46F5-A9E2-718D8BCED496-en-US-14/GUID-0C52DB00-4BF6-4F41-85B5-B76529875364.html),
+//! the number returned may not always increment by one every time it changes.
+//! In fact, testing shows that it typically increments by four every time it
+//! changes. This is true regardless of the clock rate used, as the
+//! synchronization delay scales along with the clock period.
+
+/// Items for RTIC v1.
+///
+/// TODO: This probably needs to be modernized (e.g. it does not implement
+/// half-period counting) or deprecated/removed.
+pub mod v1 {
+    use crate::rtc::{Count32Mode, Rtc};
+    use rtic_monotonic::Monotonic;
+
+    /// The RTC clock frequency in Hz.
+    pub const CLOCK_FREQ: u32 = 32_768;
+
+    /// The [`fugit`] time instant.
+    pub type Instant = fugit::Instant<u32, 1, CLOCK_FREQ>;
+    /// The [`fugit`] time duration.
+    pub type Duration = fugit::Duration<u32, 1, CLOCK_FREQ>;
+
+    impl Monotonic for Rtc<Count32Mode> {
+        type Instant = Instant;
+        type Duration = Duration;
+        unsafe fn reset(&mut self) {
+            // Since reset is only called once, we use it to enable the interrupt generation
+            // bit.
+            self.mode0().intenset().write(|w| w.cmp0().set_bit());
+        }
+
+        fn now(&mut self) -> Self::Instant {
+            Self::Instant::from_ticks(self.count32())
+        }
+
+        fn zero() -> Self::Instant {
+            Self::Instant::from_ticks(0)
+        }
+
+        fn set_compare(&mut self, instant: Self::Instant) {
+            unsafe {
+                self.mode0()
+                    .comp(0)
+                    .write(|w| w.comp().bits(instant.ticks()))
+            }
+        }
+
+        fn clear_compare_flag(&mut self) {
+            self.mode0().intflag().write(|w| w.cmp0().set_bit());
+        }
+    }
+}
+
+mod backends;
+
+use super::modes::{mode0::RtcMode0, mode1::RtcMode1, RtcMode};
+use atsamd_hal_macros::hal_macro_helper;
+
+pub mod prelude {
+    pub use super::rtc_clock;
+    pub use crate::rtc_monotonic;
+    pub use rtic_time::Monotonic;
+
+    pub use fugit::{self, ExtU32, ExtU32Ceil, ExtU64, ExtU64Ceil};
+}
```

Comment:
I don't want to start bikeshedding here, but I'm not a big fan of preludes in general. Although here, we already reexport `fugit` in `lib.rs`, so there's potential for conflict if a user does this:

```rs
use atsamd_hal::fugit::ExtU32;
use atsamd_hal::rtc::rtic::prelude::*;
```